File: Algorithms/Implementation/Utopian-tree.py
#!/bin/python3

import sys


t = int(input().strip())
for a0 in range(t):
    n = int(input().strip())
    if n%2 == 0:
        odd = n/2
        even = n/2
    else:
        odd = (n-1)/2
        even = (n+1)/2
    value = 1 + odd;
    for i in range(int(even+1)):
        value += (pow(2,i)-1)
    print(int(value))
    

File: Algorithms/Implementation/Utopiantree.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main(){
    int t;
    cin >> t;
    int odd,even,value;
    for(int a0 = 0; a0 < t; a0++){
        int n;
        cin >> n;
        if(n%2==0)
        {
            odd = n/2;
            even = n/2;
        }
        else
        {
            odd = (n-1)/2;
            even = (n+1)/2;
        }
        value = 1+ (1 * odd);
        for(int i=1; i<=even; i++)
        {
            value = value + (pow(2,i)-1);
        }
        cout<<value<<endl;
    }
    
    return 0;
}

